src/placement/jump_map.c

```diff
@@ -235,6 +235,22 @@ add_ds_shard(d_list_t *ds_list, struct pool_target *target)
 	return 0;
 }
 
+/**
+ * Free all elements in the down shard list
+ *
+ * \param[in] The ds list to be freed.
+ */
+void
+ds_list_free_all(d_list_t *used_targets_list)
+{
+	struct down_shard *d_shard, *d_tmp;
+
+	d_list_for_each_entry_safe(d_shard, d_tmp, used_targets_list, ds_list) {
```

Comment:
d_list_pop_entry() was added for this case where you're iterating a list consuming each entry.